File: Api/gestiMed2024/src/java/Clases/Paciente.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Clases;

import Listas.ListaMedico;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.Date;

/**
 *
 * @author joans
 */
public class Paciente implements Serializable{
    
    private String sip;
    private String nombre;
    private String apellidos;
    private String numContacto;
    private String nacimiento;
    private String urgencia;
    private ListaMedico medicoList;
    private Habitacion numHabitacion;
    private Tratamiento tratamiento;
    
    public Paciente() {
    }

    public Paciente(String Sip, String Nombre, String Apellidos, String NumContacto, String Nacimiento, String Urgencia) {
        this.sip = Sip;
        this.nombre = Nombre;
        this.apellidos = Apellidos;
        this.numContacto = NumContacto;
        this.nacimiento = Nacimiento;
        this.urgencia = Urgencia;
    }

    public String getSip() {
        return sip;
    }

    public void setSip(String sip) {
        this.sip = sip;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellidos() {
        return apellidos;
    }

    public void setApellidos(String apellidos) {
        this.apellidos = apellidos;
    }

    public String getNumContacto() {
        return numContacto;
    }

    public void setNumContacto(String numContacto) {
        this.numContacto = numContacto;
    }

    public String getNacimiento() {
        return nacimiento;
    }

    public void setNacimiento(String nacimiento) {
        this.nacimiento = nacimiento;
    }

    public String getUrgencia() {
        return urgencia;
    }

    public void setUrgencia(String urgencia) {
        this.urgencia = urgencia;
    }

    public ListaMedico getMedicoList() {
        return medicoList;
    }

    public void setMedicoList(ListaMedico medicoList) {
        this.medicoList = medicoList;
    }

    public Habitacion getNumHabitacion() {
        return numHabitacion;
    }

    public void setNumHabitacion(Habitacion numHabitacion) {
        this.numHabitacion = numHabitacion;
    }

    public Tratamiento getTratamiento() {
        return tratamiento;
    }

    public void setTratamiento(Tratamiento tratamiento) {
        this.tratamiento = tratamiento;
    }

    @Override
    public String toString() {
        return "Paciente{" + "sip=" + sip + ", nombre=" + nombre + ", apellidos=" + apellidos + ", numContacto=" + numContacto + ", nacimiento=" + nacimiento + ", urgencia=" + urgencia + ", medicoList=" + medicoList + ", numHabitacion=" + numHabitacion.getNumHabitacion() + ", tratamiento=" + tratamiento + '}';
    }
    
    
}

File: WPF/gestiMed2024/gestiMed2024/clases/Paciente.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Text.Json;
using CommunityToolkit.Mvvm.ComponentModel;
using gestiMed2024.listas;
using System.Text.Json.Serialization;

namespace gestiMed2024.clases
{
    class Paciente : ObservableObject
    {
        private string Apellidos;
        public string apellidos
        {
            get { return Apellidos; }
            set { SetProperty(ref Apellidos, value); }
        }

        private ListaMedicos MedicoList;
        public ListaMedicos medicoList
        {
            get { return MedicoList; }
            set { SetProperty(ref MedicoList, value); }
        }
        private string Nacimiento;
        public string nacimiento
        {
            get { return Nacimiento; }
            set { SetProperty(ref Nacimiento, value); }
        }
        private string NumContacto;
        public string numContacto
        {
            get { return NumContacto; }
            set { SetProperty(ref NumContacto, value); }
        }
        private Habitacion NumHabitacion;
        public Habitacion numHabitacion
        {
            get { return NumHabitacion; }
            set { SetProperty(ref NumHabitacion, value); }
        }
        private string Sip;
        public string sip
        {
            get { return Sip; }
            set { SetProperty(ref Sip, value); }
        }
        private string Nombre;
        public string nombre
        {
            get { return Nombre; }
            set { SetProperty(ref Nombre, value); }
        }
        private Tratamiento Tratamiento;
        public Tratamiento tratamiento
        {
            get { return Tratamiento; }
            set { SetProperty(ref Tratamiento, value); }
        }
        private string Urgencia;
        public string urgencia
        {
            get { return Urgencia; }
            set { SetProperty(ref Urgencia, value); }
        }
        
        public Paciente() { }
        public Paciente(string apellidos, string nacimiento, string nombre, string numContacto,Habitacion habitacion, string sip,Tratamiento tratamiento,string urgencia) 
        {
            this.Apellidos = apellidos;
            this.Nacimiento = nacimiento;
            this.Nombre = nombre;
            this.NumContacto = numContacto;
            this.NumHabitacion = habitacion;
            this.Sip = sip;
            this.Tratamiento = tratamiento;
            this.Urgencia = urgencia;
        }
        public Paciente(string apellidos,ListaMedicos medicos, string nacimiento, string nombre, string numContacto, Habitacion habitacion, string sip, Tratamiento tratamiento, string urgencia)
        {
            this.Apellidos = apellidos;
            this.MedicoList = medicos;
            this.Nacimiento = nacimiento;
            this.Nombre = nombre;
            this.NumContacto = numContacto;
            this.NumHabitacion = habitacion;
            this.Sip = sip;
            this.Tratamiento = tratamiento;
            this.Urgencia = urgencia;
        }
        public Paciente(string apellidos, string nacimiento, string nombre, string numContacto, string sip, string urgencia)
        {
            this.apellidos = apellidos;
            this.nacimiento = nacimiento;
            this.nombre = nombre;
            this.numContacto = numContacto;
            this.sip = sip;
            this.urgencia = urgencia;
        }
    }
}
